gbfs-validator/versions/schemas/v3.1-RC/system_pricing_plans.json

```diff
@@ -0,0 +1,205 @@
+                        "Interval in kilometers at which the rate of this segment is either reapplied indefinitely, or if defined, up until (but not including) end kilometer (added in v2.1-RC2).",
+                      "type": "integer",
+                      "minimum": 0
+                    },
+                    "end": {
+                      "description":
+                        "The kilometer at which the rate will no longer apply (added in v2.1-RC2).",
+                      "type": "integer",
+                      "minimum": 0
+                    }
+                  },
+                  "required": ["start", "rate", "interval"]
+                }
+              },
+              "per_min_pricing": {
+                "description":
+                  "Array of segments when the price is a function of time travelled, displayed in minutes (added in v2.1-RC2).",
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "start": {
+                      "description":
+                        "Number of minutes that have to elapse before this segment starts applying (added in v2.1-RC2).",
+                      "type": "integer",
+                      "minimum": 0
+                    },
+                    "rate": {
+                      "description":
+                        "Rate that is charged for each minute interval after the start (added in v2.1-RC2).",
+                      "type": "number"
+                    },
+                    "interval": {
+                      "description":
+                        "Interval in minutes at which the rate of this segment is either reapplied (added in v2.1-RC2).",
+                      "type": "integer",
+                      "minimum": 0
+                    },
+                    "end": {
+                      "description":
+                        "The minute at which the rate will no longer apply (added in v2.1-RC2).",
+                      "type": "integer",
+                      "minimum": 0
+                    }
+                  },
+                  "required": ["start", "rate", "interval"]
+                }
+              },
+              "surge_pricing": {
+                "description":
+                  "Is there currently an increase in price in response to increased demand in this pricing plan? (added in v2.1-RC2)",
+                "type": "boolean"
+              }
+            },
+            "required": [
+              "plan_id",
+              "name",
+              "currency",
+              "price",
+              "is_taxable",
+              "description"
+            ],
+            "dependencies": {
+              "reservation_price_flat_rate": { "not": { "required": ["reservation_price_per_min"] } },
+              "reservation_price_per_min": { "not": { "required": ["reservation_price_flat_rate"] } }
+            }
```

Comment:
@richfab I added this to the schema which says: either `reservation_price_flat_rate` or  `reservation_price_per_min` is present but not both of them together. Let me know if an addition like this to the schema is possible or should this be done in the form of a validation patch

Reply:
Nice improvement! Great use of the JSON schema features. I'll add it to https://github.com/MobilityData/gbfs-json-schema. Thanks 🙏

Reply:
PR is up for review: https://github.com/MobilityData/gbfs-json-schema/pull/125

Reply:
Yes, currently the schemas in https://github.com/MobilityData/gbfs-validator/ are duplicated from https://github.com/MobilityData/gbfs-json-schema/.

Open to suggestions to make sure they are always in sync.

Reply:
Two possible ways among others:

* download them from the other repo (how it's done in entur validator)
* use git submodule

Reply:
Create an issue for this: https://github.com/MobilityData/gbfs-validator/issues/186